### PR TITLE
create Product

### PR DESCRIPTION
**Pull request by DariaLavrushko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -79,6 +79,10 @@
             <artifactId>liquibase-core</artifactId>
             <version>4.25.1</version>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
```
Comment (Igor-Klein-1988) [inline pom.xml:84]: se on 27 line

File: pom.xml [hunk 2]
```diff
@@ -60,10 +60,10 @@
             <artifactId>spring-boot-starter-validation</artifactId>
         </dependency>
 
-<!--        <dependency>-->
-<!--            <groupId>org.springframework.boot</groupId>-->
-<!--            <artifactId>spring-boot-starter-security</artifactId>-->
-<!--        </dependency>-->
+        <!--        <dependency>-->
+        <!--            <groupId>org.springframework.boot</groupId>-->
+        <!--            <artifactId>spring-boot-starter-security</artifactId>-->
```
Comment (Igor-Klein-1988) [inline pom.xml:65]: do you use it?

File: src/main/java/de/allcom/controllers/ProductController.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RequiredArgsConstructor
+@RestController
+public class ProductController implements ProductApi {
+    private final ProductService productService;
+
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/controllers/ProductController.java:19]: add ` @Override`

File: src/main/java/de/allcom/controllers/ProductController.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        return productService.getAllProducts(pageRequest);
+    }
+
+    @Override
+    public ResponseEntity<ProductDto> updateProduct(UpdateProductRequestDto request) {
+        ProductDto updatedProduct = productService.updateProduct(request);
+        return new ResponseEntity<>(updatedProduct, HttpStatus.OK);
+    }
+
+    public ResponseEntity<ProductDto> createProduct(CreateProductRequestDto request) {
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/controllers/ProductController.java:31]: add ` @Override`

File: src/main/java/de/allcom/dto/product/CreateProductRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import jakarta.validation.constraints.*;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.List;
+
+@Data
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/dto/product/CreateProductRequestDto.java:14]: do you use this?

File: src/main/java/de/allcom/dto/product/CreateProductRequestDto.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package de.allcom.dto.product;
+
+import de.allcom.models.Category;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.*;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/dto/product/CreateProductRequestDto.java:5]: bad practice

File: src/main/java/de/allcom/dto/product/CreateProductRequestDto.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Schema(name = "AddProduct", description = "Data for add new product")
+public class CreateProductRequestDto {
+
+    @NotBlank(message = "Name is required")
+    @Size(min = 3, max = 50, message = "Name of the product must be between 3 and 50 characters")
+    @Schema(description = "Name of the product", example = "Regal")
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/dto/product/CreateProductRequestDto.java:23]: examples on English laungauge

File: src/main/java/de/allcom/dto/product/CreateProductRequestDto.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    private String name;
+
+    @NotBlank(message = "Description is required")
+    @Schema(description = "Description of product", example = "Sehr gute Qualitet")
+    private String description;
+
+    @Schema(description = "Weight of the product/kg", example = "3.5")
+    private Float weight;
+
+    @Schema(description = "Color of the product/kg", example = "Weiß")
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/dto/product/CreateProductRequestDto.java:33]: add `size`

File: src/main/java/de/allcom/dto/product/ProductDto.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/dto/product/ProductDto.java:11]: do you use this

File: src/main/java/de/allcom/dto/product/UpdateProductRequestDto.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+import jakarta.validation.constraints.Size;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.List;
+
+@Data
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/dto/product/UpdateProductRequestDto.java:15]: same

File: src/main/java/de/allcom/dto/product/UpdateProductRequestDto.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    @Schema(description = "Description of product", example = "Sehr gute Qualitet")
+    private String description;
+
+    @Schema(description = "Weight of the product/kg", example = "3.5")
+    private Float weight;
+
+    @Schema(description = "Color of the product/kg", example = "Weiß")
+    private String color;
+
+    @Schema(description = "Category", example = "fsda")
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/dto/product/UpdateProductRequestDto.java:40]: What?

File: src/main/java/de/allcom/dto/product/UpdateProductRequestDto.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    private String name;
+
+    @NotBlank(message = "Description")
+    @Schema(description = "Description of product", example = "Sehr gute Qualitet")
+    private String description;
+
+    @Schema(description = "Weight of the product/kg", example = "3.5")
+    private Float weight;
+
+    @Schema(description = "Color of the product/kg", example = "Weiß")
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/dto/product/UpdateProductRequestDto.java:37]: add `size`

File: src/main/java/de/allcom/exceptions/CopyPhotoException.java [hunk 13]
```diff
@@ -0,0 +1,16 @@
+package de.allcom.exceptions;
+
+import lombok.Getter;
+import org.springframework.http.HttpStatus;
+
+import java.io.IOException;
+
+@Getter
+public class CopyPhotoException extends IOException {
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/exceptions/CopyPhotoException.java]: use try/catch

File: src/main/java/de/allcom/exceptions/CopyPhotoExceptionHandler.java [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+import de.allcom.dto.StandardResponseDto;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class CopyPhotoExceptionHandler {
+
+    @ExceptionHandler(value = CopyPhotoException.class)
+    public ResponseEntity<StandardResponseDto> handleRestException(CopyPhotoException e) {
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/exceptions/CopyPhotoExceptionHandler.java]: We use RestExeption for this

File: src/main/java/de/allcom/exceptions/NotFoundException.java [hunk 15]
```diff
@@ -0,0 +1,14 @@
+package de.allcom.exceptions;
+
+import lombok.Getter;
+import org.springframework.http.HttpStatus;
+
+@Getter
+public class NotFoundException extends RuntimeException {
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/exceptions/NotFoundException.java]: delete this file

File: src/main/java/de/allcom/exceptions/NotFoundExceptionHandler.java [hunk 16]
```diff
@@ -0,0 +1,18 @@
+package de.allcom.exceptions;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/exceptions/NotFoundExceptionHandler.java]: delete this file

File: src/main/java/de/allcom/controllers/api/ProductApi.java [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.swagger.v3.oas.annotations.tags.Tags;
+import jakarta.validation.Valid;
+import org.springframework.data.domain.Page;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/controllers/api/ProductApi.java:17]: bad practice

File: src/main/java/de/allcom/controllers/ProductController.java [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+import de.allcom.dto.product.ProductDto;
+import de.allcom.dto.product.CreateProductRequestDto;
+import de.allcom.dto.product.UpdateProductRequestDto;
+import de.allcom.services.ProductService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/controllers/ProductController.java:13]: bad practice

File: src/main/java/de/allcom/models/Category.java [hunk 19]
```diff
@@ -0,0 +1,57 @@
+package de.allcom.models;
+
+import jakarta.persistence.*;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/Category.java:3]: bad practice

File: src/main/java/de/allcom/models/Category.java [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+import lombok.*;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/Category.java:4]: bad practice

File: src/main/java/de/allcom/models/Category.java [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+    @Column(nullable = false, length = 50)
+    private String nameRu;
+
+    @Column(nullable = false, length = 50)
+    private String nameDe;
+
+    @Column(nullable = false, length = 50)
+    private String nameEn;
+
+    @Column(nullable = false)
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/Category.java:31]: nullable = true

File: src/main/java/de/allcom/models/Category.java [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+    private LocalDateTime updateAt;
+
+    @Column(nullable = false)
+    private LocalDateTime createAt;
+
+    @OneToMany(mappedBy = "category")
+    private List<Product> products;
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/Category.java:44]: use https://plugins.jetbrains.com/plugin/15075-jpa-buddy

File: src/main/java/de/allcom/models/PhotoProduct.java [hunk 23]
```diff
@@ -0,0 +1,44 @@
+package de.allcom.models;
+
+import jakarta.persistence.*;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/PhotoProduct.java]: bad practice

File: src/main/java/de/allcom/models/PhotoProduct.java [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+import lombok.*;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/PhotoProduct.java]: bad practice

File: src/main/java/de/allcom/models/PhotoProduct.java [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.Objects;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@ToString
+@Entity
+@Builder
+@Table(name = "images")
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/PhotoProduct.java]: name = "product_images"

File: src/main/java/de/allcom/models/PhotoProduct.java [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+public class PhotoProduct {
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/PhotoProduct.java]: ProductImage

File: src/main/java/de/allcom/models/PhotoProduct.java [hunk 27]
```diff
@@ -0,0 +1,44 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne
+    @JoinColumn(name = "product_id", referencedColumnName = "id")
+    private Product product;
+
+    @Column(length = 1000)
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/PhotoProduct.java]: delete it

File: src/main/java/de/allcom/models/Product.java [hunk 28]
```diff
@@ -0,0 +1,47 @@
+package de.allcom.models;
+
+import jakarta.persistence.*;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/Product.java:3]: same

File: src/main/java/de/allcom/models/Product.java [hunk 29]
```diff
@@ -0,0 +1,47 @@
...
+import lombok.*;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/Product.java:4]: same

File: src/main/java/de/allcom/models/Product.java [hunk 30]
```diff
@@ -0,0 +1,47 @@
...
+@ToString
+@Builder
+@Entity
+@Table(name = "products")
+public class Product {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, length = 50)
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/Product.java:21]: min = 3?

File: src/main/java/de/allcom/models/Product.java [hunk 31]
```diff
@@ -0,0 +1,47 @@
...
+    private String name;
+
+    @Column(nullable = false)
+    private String description;
+
+    @Column
+    private Float weight;
+
+    @Column
+    private String color;
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/Product.java:31]: size?

File: src/main/java/de/allcom/models/Product.java [hunk 32]
```diff
@@ -0,0 +1,47 @@
...
+
+    @ManyToOne
+    @JoinColumn(name = "category_id", referencedColumnName = "id")
+    private Category category;
+
+    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
+    private List<PhotoProduct> photos;
+
+    public String getCategoryName() {
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/Product.java:40]: move to repository

File: src/main/java/de/allcom/models/Product.java [hunk 33]
```diff
@@ -0,0 +1,47 @@
...
+        return category.getNameDe();
+    }
+
+    public List<String> getLinksOfPhotos() {
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/models/Product.java:44]: move to repository

Comment (Igor-Klein-1988) [reply src/main/java/de/allcom/models/Product.java:44]: use jpa

File: src/main/java/de/allcom/models/Product.java (hunk 32 above)
Comment (Igor-Klein-1988) [reply src/main/java/de/allcom/models/Product.java:40]: do you use this? why?

File: src/main/java/de/allcom/repositories/PhotoProductRepository.java [hunk 34]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.List;
+
+@Repository
+public interface PhotoProductRepository extends JpaRepository<PhotoProduct, Long> {
+    List<PhotoProduct> findByProduct(Product product);
+
+    void deleteAllByLink(String link);
+    void deleteByLink(String link);
+
+    void deleteAllByProduct(Product product);
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/repositories/PhotoProductRepository.java]: use product_id

File: src/main/java/de/allcom/repositories/PhotoProductRepository.java [hunk 35]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface PhotoProductRepository extends JpaRepository<PhotoProduct, Long> {
+    List<PhotoProduct> findByProduct(Product product);
+
+    void deleteAllByLink(String link);
+    void deleteByLink(String link);
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/repositories/PhotoProductRepository.java]: my be `id`

File: src/main/java/de/allcom/repositories/PhotoProductRepository.java [hunk 36]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public interface PhotoProductRepository extends JpaRepository<PhotoProduct, Long> {
+    List<PhotoProduct> findByProduct(Product product);
+
+    void deleteAllByLink(String link);
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/repositories/PhotoProductRepository.java]: What are you deleting here?

File: src/main/java/de/allcom/services/ProductService.java [hunk 37]
```diff
@@ -0,0 +1,137 @@
...
+    private final Converters converters;
+
+    public Page<ProductDto> getAllProducts(PageRequest pageRequest) {
+        Page<Product> products = productRepository.findAll(pageRequest);
+        return products.map(converters::fromProductToProductDto);
+    }
+
+    @Transactional
+    public ProductDto createProductWithPhotos(CreateProductRequestDto request) {
+        Product product = new Product();
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/services/ProductService.java:50]: use Builder

File: src/main/java/de/allcom/services/ProductService.java [hunk 38]
```diff
@@ -0,0 +1,137 @@
...
+        product.setWeight(request.getWeight());
+        product.setCategory(request.getCategory());
+        Product savedProduct = productRepository.save(product);
+        if (request.getPhotos() != null) {
+            List<PhotoProduct> photos = uploadPhotos(request.getPhotos(), savedProduct);
+            savedProduct.setPhotos(photos);
+        } else {
+            throw new IllegalArgumentException("Null original file name");
+        }
+        productRepository.save(savedProduct);
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/services/ProductService.java:62]: Can we use one time `productRepository.save()` method?

File: src/main/java/de/allcom/services/ProductService.java [hunk 39]
```diff
@@ -0,0 +1,137 @@
...
+        return converters.fromProductToProductDto(savedProduct);
+    }
+
+    public List<PhotoProduct> uploadPhotos(List<MultipartFile> photos, Product product) {
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/services/ProductService.java:66]: move to productImageService

File: src/main/java/de/allcom/services/ProductService.java [hunk 40]
```diff
@@ -0,0 +1,137 @@
...
+            photoProduct.setProduct(product);
+            photoProductRepository.save(photoProduct);
+        }
+        return photoProducts;
+    }
+
+    @Transactional
+    public ProductDto updateProduct(UpdateProductRequestDto request) {
+        Product product = productRepository.findById(request.getId()).orElseThrow(() ->
+                new NotFoundException(HttpStatus.BAD_REQUEST, "The product is not found"));
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/services/ProductService.java:107]: use RestException

File: src/main/java/de/allcom/services/ProductService.java [hunk 41]
```diff
@@ -0,0 +1,137 @@
...
+        Category category = categoryRepository.findById(request.getCategoryId()).orElseThrow(() ->
+                new NotFoundException(HttpStatus.BAD_REQUEST, "The category is not found"));
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/services/ProductService.java:109]: use RestException

File: src/main/java/de/allcom/services/ProductService.java [hunk 42]
```diff
@@ -0,0 +1,137 @@
...
+        product.setColor(request.getColor());
+        product.setCategory(category);
+        if(!photoProductRepository.findByProduct(product).isEmpty()) {
+            List<String> altPhotos = product.getLinksOfPhotos();
+            for (String photo: altPhotos) {
+                try {
+                    Files.delete(Path.of(photo));
+                } catch (IOException e) {
+                    try {
+                        throw new CopyPhotoException(HttpStatus.BAD_REQUEST, "Error copy photo: " + photo);
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/services/ProductService.java:123]: use RestException

File: src/main/java/de/allcom/services/ProductService.java [hunk 43]
```diff
@@ -0,0 +1,137 @@
...
+                    } catch (CopyPhotoException ex) {
+                        throw new RuntimeException(ex);
```
Comment (Igor-Klein-1988) [inline src/main/java/de/allcom/services/ProductService.java:125]: same

File: src/main/resources/application.properties [hunk 44]
```diff
@@ -1,4 +1,6 @@
 spring.datasource.url=jdbc:mysql://localhost:3306/allcom_db
 spring.datasource.username=root
+spring.datasource.password=we4kl25g
```
Comment (Igor-Klein-1988) [inline src/main/resources/application.properties:3]: What?????

File: src/main/resources/db/changelog/changes/202401111215-categories-table.xml [hunk 45]
```diff
@@ -0,0 +1,32 @@
...
+                <constraints nullable="false"/>
+            </column>
+            <column name="name_de" type="varchar(50)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="name_en" type="varchar(50)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="parent_id" type="bigint">
+                <constraints nullable="false"/>
```
Comment (Igor-Klein-1988) [inline src/main/resources/db/changelog/changes/202401111215-categories-table.xml:22]: possible true

File: src/main/resources/db/changelog/changes/202401111225-insert-categories-table-dev.xml [hunk 46]
```diff
@@ -0,0 +1,536 @@
...
+            <column name="id" valueNumeric="4"/>
+            <column name="name_ru" value="Планшеты"/>
+            <column name="name_de" value="Tablets"/>
+            <column name="name_en" value="Tablets"/>
+            <column name="parent_id" valueNumeric="1"/>
+        </insert>
+        <insert tableName="categories">
+            <column name="id" valueNumeric="5"/>
+            <column name="name_ru" value="Игры"/>
+            <column name="name_de" value="Spiele / PC Games"/>
```
Comment (Igor-Klein-1988) [inline src/main/resources/db/changelog/changes/202401111225-insert-categories-table-dev.xml:40]: Games

File: src/main/resources/db/changelog/changes/202401111225-insert-categories-table-dev.xml [hunk 47]
```diff
@@ -0,0 +1,536 @@
...
+            <column name="name_en" value="Spiele / PC Games"/>
```
Comment (Igor-Klein-1988) [inline src/main/resources/db/changelog/changes/202401111225-insert-categories-table-dev.xml:41]: Spiele

File: pom.xml (hunk 1 above)
Comment (DariaLavrushko) [reply pom.xml:84]: +

File: pom.xml (hunk 2 above)
Comment (DariaLavrushko) [reply pom.xml:65]: +

File: src/main/java/de/allcom/controllers/ProductController.java (hunk 3 above)
Comment (DariaLavrushko) [reply src/main/java/de/allcom/controllers/ProductController.java:19]: +

File: src/main/java/de/allcom/controllers/ProductController.java (hunk 4 above)
Comment (DariaLavrushko) [reply src/main/java/de/allcom/controllers/ProductController.java:31]: +

File: src/main/java/de/allcom/dto/product/CreateProductRequestDto.java (hunk 5 above)
Comment (DariaLavrushko) [reply src/main/java/de/allcom/dto/product/CreateProductRequestDto.java:14]: +

File: src/main/java/de/allcom/services/ProductService.java (hunk 37 above)
Comment (DariaLavrushko) [reply src/main/java/de/allcom/services/ProductService.java:50]: ne

File: src/main/java/de/allcom/services/ProductService.java (hunk 38 above)
Comment (DariaLavrushko) [reply src/main/java/de/allcom/services/ProductService.java:62]: ne

**Pull request closed without merging**